File: AdisyonUygulamaJavaXMLsiz/app/src/main/java/com/example/adisyonuygulamajava/utils/SizeType.java
package com.example.adisyonuygulamajava.utils;

import android.view.ViewGroup;

public enum SizeType {
    MATCH_PARENT,
    WRAP_CONTENT;

    public int toLayoutParam() {
        switch (this) {
            case MATCH_PARENT:
                return ViewGroup.LayoutParams.MATCH_PARENT;
            case WRAP_CONTENT:
                return ViewGroup.LayoutParams.WRAP_CONTENT;
            default:
                return ViewGroup.LayoutParams.WRAP_CONTENT;
        }
    }
}

File: flutter_adisyon_uygulama/lib/utils/size_type.dart
enum SizeType {
  matchParent,
  wrapContent,
}

extension SizeTypeExt on SizeType {
  double? toSize() {
    switch (this) {
      case SizeType.matchParent:
        return double.infinity;
      case SizeType.wrapContent:
        return null;
    }
  }
}
